pennylane/decomposition/resources.py

```diff
@@ -350,7 +361,7 @@ def pow_resource_rep(base_class, base_params, z):
         z (int or float): the power
 
     """
-    if not isinstance(z, int) or z < 0:
+    if not qml.math.is_abstract(z) and (not _is_integer(z) or z < 0):
```

Comment:
I already forgot the Python priority queue for boolean logic... (I guess quite some dev will do so eventually some day) so could we have an explicit multi layers using brackets to distinguish here?

Reply:
Explicitly, are we dong `if not (qml.math.is_abstract(z) and (not _is_integer(z) or z < 0)):` or `if (not qml.math.is_abstract(z)) and (not _is_integer(z) or z < 0):` here?

Reply:
I think `not` always take priority. I use brackets to explicitly group `and` and `or`